src/util-ip.c

```diff
@@ -65,9 +66,9 @@ bool IPv4AddressStringIsValid(const char *str)
 
     addr[dots][alen] = '\0';
     for (int x = 0; x < 4; x++) {
-        int a = atoi(addr[x]);
-        if (a < 0 || a >= 256) {
-            SCLogDebug("out of range");
+        int a;
+        if (StringParseI32RangeCheck(&a, 10, 0, (const char *)addr[x], 0, 255) < 0) {
```

Comment:
Should we make the variable `a` an `int32_t` ?
It feels risky somehow...

Reply:
Just for my understanding, could you please explain why is this risky?

Reply:
Because `int` size may be 32 or 64 bits.
So, it may not be the same thing as the `int32_t` used by `StringParseI32RangeCheck` with its first pointer argument.
And it is better to avoid int size conversions when we can because there can be subtle bugs